### DISCUSSION TITLE
Weird behavior of some servers: cannot follow people

### DISCUSSION THREAD

**Discussion by daixtrose** [mastodon/mastodon#20162] [Q&A]
There is some interesting behavior which i do not know whether this was meant as a feature or if it is a bug. So maybe someone can enlighten me?

## Problem description

- I am using Microsoft Edge
- I am user `@markuswerle@nrw.social`, see https://nrw.social/web/@markuswerle
- Visiting users at https://mastodon.social/ and clicking on their `Follow` button yields an error message that I am not logged in, but that I can copy the link in order to paste it in a browser where I am logged in. But I am logged in already
- Note the ugly rendering, too: 
  ![image](https://user-images.githubusercontent.com/5588692/200659424-43916546-269e-44b3-b79b-33c1f148683b.png)

## Workaround for `mastodon.social`

Now if I go back to `nrw.social` I can post a message mentioning the person on `mastodon.social`. Then I can click on the user handle in my own post and choose the option to follow

## No workaround for `mast.to` and maybe other servers

- Trying to follow https://mas.to/@lefticus
- Fails as above
- Workaround from above does not work, the user handle is not clickable
   ![image](https://user-images.githubusercontent.com/5588692/200660376-dc40a12d-ba06-4d24-b914-d35ea768dae7.png)

Any ideas?

**Comment by daixtrose** [comment]
Oh crazy! The trick also works for the `mas.to` server, but only after some tries.

**Reply by PeterBindels-TomTom** [reply]
I think the cookie tracking your login to nrw.social does not move to the other servers, so they cannot know about your account or login on nrw.social. Searching for somebody in your home server should always work though, including a follow button.

**Reply by daixtrose** [reply]
The point is that with some servers there is absolutely no problem. There clicking the `Follow` button has the expected behavior. So it is either something in the settings or a bug or a version mismatch.

**Reply by daixtrose** [reply]
OK, after playing around with Mastodon for a while I am convinced this is a bug. Migrating to bug reports.

**Reply by daixtrose** [reply]
https://github.com/mastodon/mastodon/issues/20226
